src/customer/customer.controller.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Body,
+  Param,
+  UseGuards,
+  UsePipes,
+  ValidationPipe,
+  ParseUUIDPipe,
+  Query,
+  Patch,
+} from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import {
+  ApiBearerAuth,
+  ApiOperation,
+  ApiProperty,
+  ApiTags,
+} from '@nestjs/swagger';
+import { AuthGuard } from '@nestjs/passport';
+import { CreateCustomerDto } from './dto/create-customer.dto';
+import { GetUser } from 'src/common/decorators/get-user.decorator';
+import { User } from 'src/user/entities/user.entity';
+import { exceptionsFilter } from 'src/common/helpers/exceptions.helper';
+import { CustomerService } from './customer.service';
+import { isRoleCheck } from 'src/common/helpers/role-check.helper';
+import { Role } from '@prisma/client';
+import { GetCustomerFilterDto } from './dto/get-customer-filter.dto';
+import { UpdateCustomerDto } from './dto/update-customer.dto';
+
+@Controller('customer')
+@ApiTags('customer')
+@UseGuards(AuthGuard())
+@ApiBearerAuth()
+export class CustomerController {
+  constructor(private customerService: CustomerService) {}
+
+  @Post()
+  @ApiOperation({
+    summary: 'Criar um novo cliente',
+    description: 'Criar um novo cliente para a rede de franquias',
+  })
+  @UsePipes(new ValidationPipe({ whitelist: true, forbidNonWhitelisted: true }))
+  async createCustomer(
+    @Body() payload: CreateCustomerDto,
+    @GetUser() user: User,
+  ) {
+    try {
+      isRoleCheck(user.role, Role.FRANCHISEE, Role.EMPLOYEE);
+      return await this.customerService.createCustomer(payload);
+    } catch (error) {
+      exceptionsFilter(error);
+    }
+  }
+
+  @Patch(':id')
+  @ApiOperation({
+    summary: 'Atualizar um cliente',
+    description: 'Atualizar um cliente pelo id',
+  })
+  @UsePipes(new ValidationPipe({ whitelist: true, forbidNonWhitelisted: true }))
+  async update(
+    @Param('id', ParseUUIDPipe) id: string,
+    @Body() payload: UpdateCustomerDto,
+    @GetUser() user: User,
+  ) {
+    try {
+      isRoleCheck(user.role, Role.FRANCHISEE, Role.EMPLOYEE);
+      return await this.customerService.updateCustomer(id, payload);
+    } catch (error) {
+      exceptionsFilter(error);
+    }
+  }
+
+  @Get()
+  @ApiOperation({
+    summary: 'Filtro de busca de clientes',
+    description: 'Filtro de busca de clientes',
+  })
+  async findAllCustomers(
+    @Query() payload: GetCustomerFilterDto,
```

Comment:
Colocar Pipe para validar o dto do filtro
```suggestion
    @Query(ValidationPipe) payload: GetCustomerFilterDto,
```